core/new-gui/src/app/workspace/service/validation/validation-workflow.service.ts

```diff
@@ -0,0 +1,141 @@
+import { Subject } from 'rxjs/Subject';
+import { Observable } from 'rxjs/Observable';
+import { Injectable, Input } from '@angular/core';
+import { OperatorMetadataService } from './../operator-metadata/operator-metadata.service';
+import { OperatorPredicate, OperatorLink, OperatorPort} from '../../types/workflow-common.interface';
+import { OperatorSchema} from '../../types/operator-schema.interface';
+import { WorkflowActionService } from './../workflow-graph/model/workflow-action.service';
+import * as Ajv from 'ajv';
+import { isEqual } from 'lodash-es';
+@Injectable()
+export class ValidationWorkflowService {
```

Comment:
Add basic comments for this service, including
1. What this service does?
2. What library this service uses?
3. How does this service achieve the task?
4. Author.